pkg/pull/pull.go

```diff
@@ -358,7 +358,7 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 		ReportWriter:     pullOptions.ReportWriter,
 	}
 
-	if needsConfig {
+	if needsConfig && pullOptions.ConfigFile == "" {
```

Comment:
i think we need to leave this as-is. it's causing a failure in the test for required values that don't have defaults. in that case, it will skip this, try and render the helm chart, but fail because of a missing required value https://github.com/replicatedhq/kots/actions/runs/8562639222/job/23466431010#step:8:492
```
{"level":"error","ts":"2024-04-04T23:50:58Z","msg":"failed to update app from airgap bundle: failed to update app: failed to pull: failed to render upstream: failed to render helm chart airgap-smoke-test: failed to render helm chart in upstream: failed to render with helm v3: helm v3 render failed with error: execution error at (airgap-smoke-test/templates/deployment.yaml:25:24): myNewRequiredValue is required!"}
```



is there a reason we need it for `kots template`?

Reply:
Many thanks @cbodonnell! I initially thought that this logic does not include the config values in the check but later found out that it is. I removed the incorrect check.